homeassistant/components/matter/climate.py

```diff
@@ -99,23 +101,55 @@ def hvac_action(self) -> HVACAction | None:
         if running_state := self.get_matter_attribute_value(
             clusters.Thermostat.Attributes.ThermostatRunningState
         ):
-            if running_state == 0:
-                return HVACAction.HEATING
-            if running_state == 1:
-                return HVACAction.COOLING
-            if running_state == 2:
-                return HVACAction.FAN
-        if running_mode := self.get_matter_attribute_value(
-            clusters.Thermostat.Attributes.ThermostatRunningMode
-        ):
-            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kCool:
-                return HVACAction.COOLING
-            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kHeat:
-                return HVACAction.HEATING
-            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kOff:
-                return HVACAction.OFF
+            match running_state:
+                case 1 | 8:
+                    return HVACAction.HEATING
+                case 2 | 16:
+                    return HVACAction.COOLING
+                case 4 | 32 | 64:
+                    return HVACAction.FAN
+                case _:
+                    return HVACAction.OFF
         return None
 
+    @property
+    def min_temp(self) -> float | None:
+        """Return the minimum temperature."""
+        match self.hvac_mode:
+            case HVACMode.COOL:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.AbsMinCoolSetpointLimit
+                )
+            case HVACMode.HEAT:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.AbsMinHeatSetpointLimit
+                )
+            case HVACMode.HEAT_COOL:
+                return self._get_temperature_in_degrees(
```

Comment:
This becomes very verbose as we're repeating the temperature calculation. Can we move the temperature calculation out of the match block to the end of the body?

Reply:
these are different attributes (cool vs heat)